### ISSUE TITLE
Push user registration more towards server

### ISSUE THREAD

**Issue by timroes** [airbytehq/airbyte#10495]
Currently most parts of the process of user registration is done in the frontend and the actual user in our DB is only created once the mail verification is done. That also means we currently need to store all additional user informatin "in the frontend" (or as meta data attached in Firebase), while this process is still ongoing (see https://github.com/airbytehq/airbyte-cloud/issues/853 for more context).

We should try to move more of that logic to the airbyte-server:

* The frontend should create a new user in the airbyte-server immediately on registration storing all the user information in it.
* The backend should trigger the user generation and mail verification in firebase.
* When the mail verification link is opened, the FE should call an API in the backend for that, which itself calls the firebase mail verification API (currently the FE calls the mail verification directly).
* Most likely we'd want a periodic job on the backend cleaning up users that never verify their mail addresses.

**Comment by andyjih** [comment]
I put together a [whimsical](https://whimsical.com/Dp6nnCM97pse499UZuDJ5u) flowchart here to capture what I think we're discussing. That said, I made a few assumptions about I think might be happening server-side and in firebase, so please provide comments/corrections!

**Comment by davinchia** [comment]
@timroes my general understanding of this is we want to slim down what the FE is responsible when doing auth. Is that accurate?

**Comment by timroes** [comment]
@davinchia that is correct. I believe we want less of the auth logic in the FE than we currently have. That said it's working atm and I don't see any real blockers with the given implementation. I'd treat this more like a tech-debt that would be nice to fix rather than something we need to address urgently.

**Comment by davinchia** [comment]
@timroes agreed. was just confirming my understanding. thanks for following up.
